File: app/src/main/java/br/com/menuvem/lojista/domain/model/HistoricoPreco.kt
package br.com.menuvem.lojista.domain.model

import java.time.LocalDateTime

/**
 * Modelo de domínio: Registro histórico de preço de um insumo.
 *
 * @param preco           Preço pago por unidade de COMPRA do insumo.
 * @param listaComprasId  Lista de compras de origem deste registro.
 */
data class HistoricoPreco(
    val id: Long = 0,
    val insumoId: Long,
    val preco: Double,
    val data: LocalDateTime = LocalDateTime.now(),
    val listaComprasId: Long,
    val listaComprasNome: String = ""
)

File: lib/domain/model/historico_preco.dart
/// Registro histórico de preço de um insumo.
///
/// [preco]: preço pago por unidade de COMPRA do insumo.
/// [listaComprasId]: lista de compras de origem deste registro.
class HistoricoPreco {
  final int id;
  final int insumoId;
  final double preco;
  final DateTime data;
  final int listaComprasId;
  final String listaComprasNome;

  HistoricoPreco({
    this.id = 0,
    required this.insumoId,
    required this.preco,
    DateTime? data,
    required this.listaComprasId,
    this.listaComprasNome = '',
  }) : data = data ?? DateTime.now();
}
